### PR TITLE
Show payment entry modal on invoice page

### PR DESCRIPTION
**Pull request by apoorv1316**
What
- Show payment entry modal onClick on the "Mark as paid" option on the Invoice page

Why
- For better UX. Earlier it used to redirect to the payments page

Loom: 
https://www.loom.com/share/45ca4e3552ad40f191fb9373379727d6

### CODE REVIEW

**Review by supriya3105 — APPROVED** [review]
Looks good . 
There is a minor change required. The first field on the modal is displaying the client name whereas it should display the invoice number as well. 
However that is an existing issue which we can fix in separate PR.

File: app/javascript/src/components/Invoices/Invoice/MarkInvoicePaidModal.tsx [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+          label="Notes (optional)"
+          maxRows={12}
+          name="NotesOptional"
+          rows={5}
+          value={note}
+          onChange={e => setNote(e.target.value)}
+        />
+      </div>
+      <div className="actions mx-auto mt-4 mb-4 w-full">
+        <button
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/Invoice/MarkInvoicePaidModal.tsx:221]: If possible use the `Button` component from `StyledComponents`

File: app/javascript/src/components/Invoices/Invoice/InvoiceActions.tsx [hunk 2]
```diff
@@ -70,6 +73,16 @@ const InvoiceActions = ({
             }}
           />
         )}
+        {showInvoicePaymentModal && (
+          <MarkInvoiceAsPaidModal
+            baseCurrency={invoice.company.currency}
```
[the remark this answers is not in the document]
Comment (apoorv1316) [reply app/javascript/src/components/Invoices/Invoice/InvoiceActions.tsx]: done

File: app/javascript/src/components/Invoices/Invoice/MarkInvoicePaidModal.tsx [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+          <div className="mt-1" id="invoice" ref={wrapperSelectRef}>
+            <CustomInputText
+              disabled
+              id="invoice"
+              inputBoxClassName="form__input block w-full appearance-none bg-white p-4 text-base h-12 focus-within:border-miru-han-purple-1000 border-miru-gray-1000"
+              label="Invoice"
+              labelClassName="absolute top-0.5 left-1 h-6 z-1 origin-0 bg-white p-2 text-base font-medium duration-300 text-miru-dark-purple-200"
+              name="invoice"
+              type="text"
+              value={invoice.client.name}
```
[the remark this answers is not in the document]
Comment (apoorv1316) [reply app/javascript/src/components/Invoices/Invoice/MarkInvoicePaidModal.tsx]: done

File: app/javascript/src/components/Invoices/Invoice/MarkInvoicePaidModal.tsx (hunk 1 above)
Comment (apoorv1316) [reply app/javascript/src/components/Invoices/Invoice/MarkInvoicePaidModal.tsx]: done

**Pull request merged**